File: 0035-search-insert-position/0035-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        auto it=find(nums.begin(),nums.end(),target);
        if(it!=nums.end())
            return it-nums.begin();
        for(int i=0;i<nums.size();i++){
            if(nums[i]>target)
                return i;
        }
        return nums.size();
    }
};

File: 0035-search-insert-position/0035-search-insert-position.cs
public class Solution {
    public int SearchInsert(int[] nums, int target) {
        int l=0,h=nums.Length-1,m;
        while(l<=h){
            m=l+(h-l)/2;
            if(nums[m]==target)return m;
            else if(nums[m]<target)l=m+1;
            else if(nums[m]>target)h=m-1;
        }
        return l;
    }
}